packages/typescript/internal/ts_project.bzl

```diff
@@ -540,6 +542,26 @@ def ts_project_macro(
 
         args: List of strings of additional command-line arguments to pass to tsc.
 
+        transpiler: What tool to run that produces the JavaScript outputs.
+            By default, this is the string `tsc` which means to produce `.js` outputs
+            in the same action that does the type-checking to produce `.d.ts` outputs.
+            This is the simplest configuration, however `tsc` is slower than alternatives.
+            It also means developers must wait for the type-checking in the developer loop.
+
+            In theory, Persistent Workers (via the `supports_workers` attribute) remedies the
+            slow compilation time, however it adds additional complexity because the worker process
+            can only see one set of dependencies, and so it cannot be shared between different
+            `ts_project` rules. That attribute is documented as experimental, and may never graduate
+            to a better support contract.
+
+            Instead, you can pass a rule or macro that accepts these arguments:
+            `(name, srcs, js_outs, map_outs, args, data, tags, visibility)`
+
+            The rules_nodejs authors believe that [SWC](https://swc.rs) is a great choice.
+
+        transpiler_args: if the `transpiler` attribute is a rule or macro, then this list
```

Comment:
what about making this a dict to support arbitrary transpiler attrs such as `swcrc`?

```
transpiler_attrs = {
    "args": [...],
    "swcrc": "...",
}
```

Reply:
you can check the type of a param in a macro so in theory you could overload `transpiler_args` to support either a list (passed to args) or a dict (to support multiple attrs)

Reply:
I did think about swcrc specifically, and a common case could be to pass it as a config setting `bazel build --@aspect_rules_swc//:rc=.swcrc` so it applies as the default for your whole build.
But of course someone will have the use case of different swcrc for two different folders of code, so yes I could see this being needed.

Reply:
People who are using this attributes  are most likely going to write a wrapper with defaults. Can't they just write a wrapper around swc that is passed to ts project which invokes swc with defaults? Does this sound too complex?

Reply:
just changed it to transpiler_kwargs for simplicity